tests/linalg_test.py

```diff
@@ -1619,6 +1619,39 @@ def testToeplitzConstructionWithKnownCases(self):
       [2, 1, 4, 5],
       [3, 2, 1, 4]], dtype=np.float32))
 
+  @jtu.sample_product(
+    size=[1, 2, 3, 10, 100],
+    dtype=float_types + complex_types)
+  def testNullSpace(self, size, dtype):
+    rng = jtu.rand_default(self.rng())
+    shape = (2, size)
+    X = rng(shape, dtype)
+    args_maker = lambda: [X]
+    self._CheckAgainstNumpy(jtu.promote_like_jnp(osp.linalg.null_space),
+                            jsp.linalg.null_space, args_maker)
```

Comment:
Add here
```python
if size is not None:
  self._CompileAndCheck(jsp.linalg.null_space, args_maker)
```
This would cover the jit-compilation checks, and we may be able to delete everything below.

Reply:
see my comment above

Reply:
And I did make sure to test that compilation with `size=None` returns a meaningful error